services/site-scout/launch-worker.sh

```diff
@@ -56,23 +56,42 @@ fi
 update-ec2-status "Setup: fetching build"
 
 # select build
-echo "Build types: ${BUILD_TYPES}"
-BUILD_SELECT_SCRIPT="import random;print(random.choice(str.split('${BUILD_TYPES}')))"
-build="$(python3 -c "$BUILD_SELECT_SCRIPT")"
-# download build
-case $build in
-  asan32)
-    # TEMPORARY workaround for frequent OOMs
-    export ASAN_OPTIONS=malloc_context_size=20:rss_limit_heap_profile=false:max_malloc_fill_size=4096:quarantine_size_mb=64
-    python3 -m fuzzfetch -n build --fuzzing --asan --cpu x86
-    ;;
-  debug32)
-    python3 -m fuzzfetch -n build --fuzzing --debug --cpu x86
-    ;;
-  *)
-    python3 -m fuzzfetch -n build --fuzzing "--$build"
-    ;;
-esac
+if [[ -n "$COVERAGE" ]]; then
+  export COVERAGE_FLAG="--coverage"
+  python3 -m fuzzfetch -n build --fuzzing --coverage
+  export ARTIFACT_ROOT="https://community-tc.services.mozilla.com/api/index/v1/task/project.fuzzing.coverage-revision.latest/artifacts/public"
+  SOURCE_URL="$(resolve-url "$ARTIFACT_ROOT/source.zip")"
+  export SOURCE_URL
+
+  if [[ -z "$REVISION" ]]; then
+    REVISION="$(retry-curl --compressed "$ARTIFACT_ROOT/coverage-revision.txt")"
+  fi
+  export REVISION
+
+  export GCOV_PREFIX="$HOME/build"
+  GCOV_PREFIX_STRIP="$(grep pathprefix "${TARGET_BIN}.fuzzmanagerconf" | grep -E -o "/.+$" | tr -cd '/' | wc -c)"
+  export GCOV_PREFIX_STRIP
+else
+  export COVERAGE_FLAG=""
+  echo "Build types: ${BUILD_TYPES}"
+  BUILD_SELECT_SCRIPT="import random;print(random.choice(str.split('${BUILD_TYPES}')))"
+  build="$(python3 -c "$BUILD_SELECT_SCRIPT")"
+  # download build
+  case $build in
+    asan32)
+      # TEMPORARY workaround for frequent OOMs
+      export ASAN_OPTIONS=malloc_context_size=20:rss_limit_heap_profile=false:max_malloc_fill_size=4096:quarantine_size_mb=64
+      python3 -m fuzzfetch -n build --fuzzing --asan --cpu x86
+      ;;
+    debug32)
+      python3 -m fuzzfetch -n build --fuzzing --debug --cpu x86
+      ;;
+    *)
+      python3 -m fuzzfetch -n build --fuzzing "--$build"
+      ;;
+  esac
+fi
+TARGET_BIN="./build/firefox"
```

Comment:
You're using `TARGET_BIN` on line 72 so it should be defined before the `if` maybe.